specs/SharedBuffer.md

```diff
@@ -0,0 +1,310 @@
+  /// the buffer from that separate process.
+  [propget] HRESULT Handle([out, retval] HANDLE* value);
+  
+  /// Release the backing shared memory. The application should call this API when no
+  /// access to the buffer is needed any more, to ensure that the underlying resources
+  /// are released timely even if the shared buffer object itself is not released due to
+  /// some leaked reference.
+  /// After the shared buffer is closed, accessing properties of the object will fail with
+  /// `HRESULT_FROM_WIN32(ERROR_INVALID_STATE)`. Operations like Read or Write on the IStream objects
+  /// returned from `GetStream` will fail with `HRESULT_FROM_WIN32(ERROR_INVALID_STATE)`.
+  /// `PostSharedBufferToScript` will also fail with `HRESULT_FROM_WIN32(ERROR_INVALID_STATE)`.
+  ///
+  /// The script code should call `chrome.webview.releaseSharedBuffer` with
+  /// the shared buffer as the parameter to release underlying resources as soon
+  /// as it does not need access the shared buffer any more.
+  /// When script tries to access the buffer after calling `chrome.webview.releaseSharedBuffer`,
+  /// JavaScript `TypeError` exception will be raised complaining about accessing a detached ArrayBuffer,
+  /// the same exception when trying to access a transferred ArrayBuffer.
+  ///
+  /// Closing the buffer object on native side doesn't impact access from Script and releasing the buffer
+  /// from script doesn't impact access to the buffer from native side.
+  /// The underlying shared memory will be released by the OS when both native and script side releases the buffer.
+  HRESULT Close();
+}
+
+interface ICoreWebView2Staging7 : IUnknown {
+  /// Share a shared buffer object with script of the main frame in the WebView.
+  /// The script will receive a `SharedBufferReceived` event from chrome.webview.
+  /// The event arg for that event will have the following properties:
+  ///   `sharedBuffer`: an ArrayBuffer object with the backing content from the shared buffer.
+  ///   `data`: an object representing `additionalDataAsJson`. This property will be `undefined`
+  ///           if `additionalDataAsJson` is nullptr or empty string.
+  ///   `source`: with a value set as `chrome.webview` object.
+  /// If `isReadOnlyToScript` is true, the script will only have read access to the buffer.
+  /// If the script tries to modify the content in a read only buffer, it will cause an access violation
+  /// in WebView renderer process and crash the renderer process.
+  /// If the shared buffer is already closed, the API will fail with HRESULT_FROM_WIN32(ERROR_INVALID_STATE).
+  /// 
+  /// The script code should call `chrome.webview.releaseSharedBuffer` with
+  /// the shared buffer as the parameter to release underlying resources as soon
+  /// as it does not need access to the shared buffer any more.
+  ///
+  /// Sharing a buffer to script has security risk. You should only share buffer with trusted site.
```

Comment:
It would be good to elaborate.

I think there's two security issues here right? 1 Normal information disclosure and 2 if the data is shared with script as not-read only then script may change the data in a manner the native code doesn't except? Is that right?

Reply:
add more descriptions